federatedrec/factorization_machine/homo_factorization_machine/homo_fm_arbiter.py

```diff
@@ -0,0 +1,141 @@
+                                                                       suffix=suffix)
+
+                if model_shape == -1:
+                    if self.init_param_obj.fit_intercept:
+                        model_shape = int((len(merged_model._weights) - 1) / (embed_size + 1))
+                    else:
+                        model_shape = int(len(merged_model._weights) / (embed_size + 1))
+
+                    # Initialize the model
+                    fit_intercept = False
+                    if self.init_param_obj.fit_intercept:
+                        fit_intercept = True
+                        self.init_param_obj.fit_intercept = False
+                    w_ = self.initializer.init_model(model_shape, init_params=self.init_param_obj)
+                    embed_ = self.initializer.init_model([model_shape, self.init_param_obj.embed_size],
+                                                         init_params=self.init_param_obj)
+                    self.model_weights = \
+                        FactorizationMachineWeights(w_, embed_, fit_intercept=fit_intercept)
+
+                total_loss = self.aggregator.aggregate_loss(-1, suffix)
+                self.callback_loss(self.n_iter_, total_loss)
+                self.loss_history.append(total_loss)
+                if self.use_loss:
+                    converge_var = total_loss
+                else:
+                    converge_var = np.array(merged_model.unboxed)
+
+                self.is_converged = self.aggregator.send_converge_status(self.converge_func.is_converge,
+                                                                         (converge_var,),
+                                                                         suffix=(self.n_iter_,))
+                LOGGER.info("n_iters: {}, total_loss: {}, converge flag is :{}".format(self.n_iter_,
+                                                                                       total_loss,
+                                                                                       self.is_converged))
+                if self.is_converged:
+                    break
+                # self.model_weights = FactorizationMachineWeights(merged_model.coef_, merged_model.embed_,
+                #                                                  self.model_param.init_param.fit_intercept)
+                merged_model._weights = np.array(merged_model._weights)
+                self.model_weights.update(merged_model)
+                if self.header is None:
+                    self.header = ['x' + str(i) for i in
+                                   range(model_shape)]
+
+            validation_strategy.validate(self, self.n_iter_)
+            self.n_iter_ += 1
+
+        LOGGER.info("Finish Training task, total iters: {}".format(self.n_iter_))
+
+    def predict(self, data_instantces=None):
+        LOGGER.info(f'Start predict task')
+
+    def run(self, component_parameters=None, args=None):
```

Comment:
The run function is no need to be rewrite in arbiter now. Suggest to remove and test it again.